### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by gmorales**
New:
- Unit tests.
- Jacoco configuration.

### CODE REVIEW

File: app/src/main/java/com/techtalk/mvp/mvpandroid/domain/interactor/LoginInteractorImpl.java [hunk 1]
```diff
@@ -13,14 +13,20 @@
...
+
+    public LoginInteractorImpl(OnLoginFinishedListener listener) {
+        this.listener = listener;
+    }
+
     @Override
-    public void login(final String username, final String password, final OnLoginFinishedListener listener) {
+    public void login(final String username, final String password) {
         // Acá llamaríamos a una API de logueo, Base de datos, caché (shared preferences)
         if (TextUtils.isEmpty(username) || !username.equals(GLOBANT_USERNAME)) {
```
Comment (AnDevCba) [inline app/src/main/java/com/techtalk/mvp/mvpandroid/domain/interactor/LoginInteractorImpl.java]: Remove GLOBANT prefix, in order to be generic.

File: app/src/test/java/com/techtalk/mvp/mvpandroid/domain/interactor/LoginInteractorImplTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    @Test
+    public void loginWithIncorrectUsernameShouldReturnUsernameError() {
+        interactor.login("guest", "Globant");
+
+        verify(listener).onUsernameError();
+    }
+
+    @Test
+    public void loginWithNullUsernameShouldReturnUsernameError() {
+        interactor.login(null, "Globant");
```
Comment (AnDevCba) [inline app/src/test/java/com/techtalk/mvp/mvpandroid/domain/interactor/LoginInteractorImplTest.java:57]: Rename for a generic password.

**Review by AnDevCba — APPROVED** [review]
LG

**Comment by gmorales** [comment]
@AnDevCba PR actualizado.

**Pull request merged**